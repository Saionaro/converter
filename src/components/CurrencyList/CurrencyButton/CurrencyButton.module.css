.root {
  background-color: var(--color-white);
  border: 1px solid var(--color-primary);
  border-radius: 12px;
  line-height: 24px;
  padding: 0px 12px;
  font-size: 10px;
  transition-property: opacity, background-color;
  transition-duration: var(--time-interaction);
}

.root:hover:not(:disabled),
.root:focus:not(:disabled) {
  cursor: pointer;
  background-color: var(--color-primary-light);
  outline: none;
}

.root:active:not(:disabled) {
  opacity: 0.6;
  color: var(--color-black);
}

.active {
  color: var(--color-white);
  background-color: var(--color-primary);
  font-weight: bold;
}

.amount {
  display: inline-block;
  line-height: 24px;
  padding-left: 10px;
  margin-left: 10px;
  border-left: 1px solid var(--color-primary);
}

@media screen and (max-width: 700px) {
  .amount {
    display: none;
  }
}
