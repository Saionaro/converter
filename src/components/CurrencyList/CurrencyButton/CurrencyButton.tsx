import React, { useCallback } from "react";
import cn from "classnames";
import { Currency } from "src/common-types";
import { formatters } from "src/utils/format";

import st from "./CurrencyButton.module.css";

export const TEST_ID = "currency-button";

interface Props {
  currency: Currency;
  onClick: (curr: Currency) => void;
  active?: boolean;
  disabled?: boolean;
  amount?: string;
}

export function CurrencyButton({
  currency,
  onClick,
  amount,
  disabled,
  active,
}: Props) {
  const handleClick = useCallback(() => onClick(currency), [currency, onClick]);

  return (
    <button
      onClick={handleClick}
      className={cn(st.root, active && st.active)}
      disabled={disabled}
      data-testid={TEST_ID}
    >
      {currency}
      {!!amount && (
        <span className={st.amount}>{formatters[currency](amount)}</span>
      )}
    </button>
  );
}
