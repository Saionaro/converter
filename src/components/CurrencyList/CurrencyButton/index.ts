export { CurrencyButton, TEST_ID } from "./CurrencyButton";
