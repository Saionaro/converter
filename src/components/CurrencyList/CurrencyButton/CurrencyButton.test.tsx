import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { CurrencyButton, TEST_ID } from "./CurrencyButton";

describe("components/CurrencyButton", () => {
  test("fit snapshot", () => {
    const { container } = render(
      <CurrencyButton currency="USD" onClick={jest.fn()} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test("invokes callback with currency id", () => {
    const mockCallback = jest.fn();
    const { getByTestId } = render(
      <CurrencyButton currency="USD" onClick={mockCallback} />
    );

    const button = getByTestId(TEST_ID);
    fireEvent.click(button);

    expect(mockCallback).toBeCalledTimes(1);
    expect(mockCallback).toBeCalledWith("USD");
  });
});
