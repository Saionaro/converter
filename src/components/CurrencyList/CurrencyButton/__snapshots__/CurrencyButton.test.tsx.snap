// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CurrencyButton fit snapshot 1`] = `
<button
  class=""
  data-testid="currency-button"
>
  USD
</button>
`;
